### PR TITLE
Use tree.replace instead of tree.mutate to change a node's field

### PR DESCRIPTION
**Pull request by stellarhoof**
With this we'll change the exampleType component correctly as well as take into account custom default node props

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>You should add at least 1 more reviewer to the PR</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>You reduced the total lines of code! Awesome! :+1:</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 57cc581ff9f4084e3312ce079400246257354425
</p>

File: src/FilterAdder.js [hunk 1]
```diff
@@ -2,21 +2,13 @@ import _ from 'lodash/fp'
...
 let getGroupFields = node => _.map('field', _.getOr([], 'children', node))
 
-let FilterAdder = ({
-  tree,
-  node,
-  path,
-  fields,
-  Picker,
-  uniqueFields,
-  defaultNodeProps = DefaultNodeProps,
```
Comment (stellarhoof) [inline src/FilterAdder.js]: @daedalus28 Do we want this? Nothing uses it.

Comment (daedalus28) [reply src/FilterAdder.js]: also even if it does not, contexture-react shouldn't dictate that's the only way to add default node props

File: src/utils/search.js [hunk 2]
```diff
@@ -18,13 +18,15 @@ export let newNodeFromType = _.curry((type, fields, node) => ({
...
 }))
 
-// In the future, we will have logic here to handle intelligently updating
-// a node's types when its field is mutated. When that happens, we'll also
-// need to pass the `fields` schema to look up typeOptions for the current
-// and future field. Could also be a node transformer like newNodeFromType
-// if we decide to use Tree.replace() for fields instead of Tree.mutate().
-export let changeNodeField = (Tree, node, field) => {
-  Tree.mutate(node.path, { field })
+export let newNodeFromField = ({ key, field, fields, ...rest }) => {
```
Comment (michsa) [inline src/utils/search.js]: personally, I don't prefer this style of argument passing unless it's really necessary - I'd probably just have done `(field, fields, rest = {})`, with `key` in the rest object (or not at all, see the next comment)

it's really just a matter of taste, though, so it's up to you 🙂

File: src/FilterAdder.js (hunk 1 above)
Comment (stellarhoof) [reply src/FilterAdder.js]: > contexture-react shouldn't dictate that's the only way to add default node props

Adding it back to merge this PR, but this flexibility is only supported here. We just go with a vanilla `defaultNodeProps` everywhere else

Comment (stellarhoof) [reply src/FilterAdder.js]: > contexture-react shouldn't dictate that's the only way to add default node props

Why here and not in all the other places we're using it? This is the only place where we offer overriding it and we're not using it anyway.

Comment (daedalus28) [reply src/FilterAdder.js]: ok, we can always add this back later if need be then

**Pull request merged**